nixos/modules/services/backup/tarsnap.nix

```diff
@@ -242,9 +254,11 @@ in
 
     systemd.services."tarsnap@" = {
       description = "Tarsnap archive '%i'";
-      requires    = [ "network.target" ];
+      requires    = [ "network-online.target" ];
+      after       = [ "network-online.target" ];
 
-      path = [ pkgs.tarsnap pkgs.coreutils ];
+      path = [ pkgs.tarsnap pkgs.coreutils pkgs.iputils ];
+      preStart = "/run/current-system/sw/bin/sh -c 'while ! ping -q -c 1 betatest-server.tarsnap.com &> /dev/null; do true; done'";
```

Comment:
To add to this: for persistent timers, it seems systemd only keeps track of whether the service was triggered, not whether it completed successfully. For backups, I suppose the latter is what we actually want ... I'm not quite sure if this hack even accomplishes this, though (i.e, whether systemd waits until the main process starts before counting the service as "triggered").